source/Stargate-API-Skeleton/StargateApplication.class.st

```diff
@@ -169,6 +186,32 @@ StargateApplication >> configureGlobalErrorHandlerIn: api [
 	] unless: self isDebugModeEnabled
 ]
 
+{ #category : #'private - activation/deactivation' }
+StargateApplication >> configureHTTPClientLogging [
+
+	self stargateConfiguration logClientEvents then: [ 
+		ZnLogEvent announcer when: ZnClientTransactionEvent do: [ :event | 
+			LaunchpadLogRecord
+				emitStructuredDebuggingInfo: 'HTTP Client event received'
+				with: [ :data | self dumpDataOf: event on: data ] ] ]
+]
+
+{ #category : #'private - activation/deactivation' }
+StargateApplication >> configureHTTPServerLogging [
+
+	self stargateConfiguration logServerEvents then: [ 
+		ZnLogEvent announcer
+			when: ZnServerTransactionEvent , ZnSimplifiedServerTransactionEvent
+			do: [ :event | 
+				LaunchpadLogRecord
+					emitStructuredDebuggingInfo: 'HTTP Server event received'
+					with: [ :data | self dumpDataOf: event on: data ] ] ].
+
+	ZnLogEvent announcer
+		when: ZnServerHandlerErrorEvent
+		do: [ :event | LaunchpadLogRecord emitError: event exception messageText ]
```

Comment:
We can send directly a message to the event here, something like emitAsLogRecord, so we don't need to expose the internals of the event.

Reply:
And we can test more easily different combinations of event data